api/src/search/backend/load_agencies_to_index.py

```diff
@@ -72,25 +77,55 @@ def fetch_agencies(self) -> Sequence[Agency]:
         """
         return self.db_session.execute(select(Agency).options(selectinload("*"))).scalars().all()
 
-    def load_agencies(self, agencies: Sequence[Agency]) -> None:
-        logger.info("Loading agencies...")
-        active_agency_subquery = (
-            _construct_active_inner_query(Agency.agency_id)
-            .union(_construct_active_inner_query(Agency.top_level_agency_id))
+    def _get_agencies_by_status(self, status: OpportunityStatus) -> set[Agency]:
+        """Fetch agencies based on the status."""
+        agencies_subquery = (
+            _construct_active_inner_query(Agency.agency_id, [status])
+            .union(_construct_active_inner_query(Agency.top_level_agency_id, [status]))
             .subquery()
         )
 
-        agency_id_stmt = select(active_agency_subquery).distinct()
-        active_agencies = set(self.db_session.execute(agency_id_stmt).scalars())
+        agency_id_stmt = select(agencies_subquery).distinct()
+        return set(self.db_session.execute(agency_id_stmt).scalars())
+
+    def load_agencies(self, agencies: Sequence[Agency]) -> None:
+        logger.info("Loading agencies...")
+
+        # Get agencies by opportunity status
+        posted_agencies = self._get_agencies_by_status(OpportunityStatus.POSTED)
+        forecasted_agencies = self._get_agencies_by_status(OpportunityStatus.FORECASTED)
+        closed_agencies = self._get_agencies_by_status(OpportunityStatus.CLOSED)
+        archived_agencies = self._get_agencies_by_status(OpportunityStatus.ARCHIVED)
 
         agencies_json = []
         for agency in agencies:
             logger.info(
                 "Preparing agency for upload to search index",
                 extra={"agency_id": agency.agency_id, "agency_code": agency.agency_code},
             )
+
             agency_json = SCHEMA.dump(agency)
-            agency_json["has_active_opportunity"] = agency.agency_id in active_agencies
+
+            opportunity_statuses = []
+
+            if agency.agency_id in posted_agencies:
+                opportunity_statuses.append(OpportunityStatus.POSTED.value)
+
+            if agency.agency_id in forecasted_agencies:
+                opportunity_statuses.append(OpportunityStatus.FORECASTED.value)
+
+            if agency.agency_id in closed_agencies:
+                opportunity_statuses.append(OpportunityStatus.CLOSED.value)
+
+            if agency.agency_id in archived_agencies:
+                opportunity_statuses.append(OpportunityStatus.ARCHIVED.value)
```

Comment:
OpportunityStatus is a string and an enum, should be able to exclude value
```suggestion
            if agency.agency_id in posted_agencies:
                opportunity_statuses.append(OpportunityStatus.POSTED)

            if agency.agency_id in forecasted_agencies:
                opportunity_statuses.append(OpportunityStatus.FORECASTED)

            if agency.agency_id in closed_agencies:
                opportunity_statuses.append(OpportunityStatus.CLOSED)

            if agency.agency_id in archived_agencies:
                opportunity_statuses.append(OpportunityStatus.ARCHIVED)
```